test/compilable/testcolor.sh

```diff
@@ -0,0 +1,46 @@
+set -eo pipefail
+
+compare()
+{
+    local actual expected
+    actual="$1"
```

Comment:
I recommend assigning in the declaration.

Reply:
Assigning in declaration discards exit code, shellcheck complains